### PR TITLE
Upload bench output as build artifacts

### PR DESCRIPTION
**Pull request by pgarg66**

### CODE REVIEW

**Comment by pgarg66** [comment]
https://github.com/solana-labs/solana/issues/1456

File: src/bin/upload-perf.rs [hunk 1]
```diff
@@ -1,14 +1,46 @@
...
 
+fn get_last_metrics(
+    url: &str,
+    metric: &str,
+    db: &str,
+    name: &str,
+    branch: &str,
+) -> Result<String, String> {
+    let query = format!(
+        "{}&q=SELECT last(\"{}\") FROM \"{}\".\"autogen\".\"{}\" WHERE \"branch\"='{}'",
```
Comment (mvines) [inline src/bin/upload-perf.rs]: Try using a raw string to eliminate the double quote escaping

File: src/bin/upload-perf.rs [hunk 2]
```diff
@@ -18,33 +50,89 @@ fn main() {
         Ok(file) => file,
     };
 
+    let upload_flag = &args[2];
```
Comment (mvines) [inline src/bin/upload-perf.rs]: nit: How about flipping the order of the branch and upload_flag arguments, then make upload_flag optional (where missing == "no-upload")

File: src/bin/upload-perf.rs [hunk 3]
```diff
@@ -18,33 +50,89 @@ fn main() {
...
+
     let git_output = Command::new("git")
         .args(&["rev-parse", "HEAD"])
         .output()
         .expect("failed to execute git rev-parse");
     let git_commit_hash = String::from_utf8_lossy(&git_output.stdout);
     let trimmed_hash = git_commit_hash.trim().to_string();
-    println!("uploading hash: {}", trimmed_hash);
+
+    let host =
```
Comment (mvines) [inline src/bin/upload-perf.rs]: Let's move this logic and a generic query function (that `get_last_metrics()` invokes) into src/metrics.rs.  This way we can easily re-use the same default database/password in that file.

Also please change your InfluxDB password, the world now knows it ;)

**Comment by pgarg66** [comment]
@mvines , the latest patch should address your review comments.

File: src/metrics.rs [hunk 4]
```diff
@@ -177,6 +179,21 @@ pub fn submit(point: influxdb::Point) {
     agent.submit(point);
 }
 
+pub fn query(q: &str) -> Result<String, String> {
+    let host =
+        env::var("INFLUX_HOST").unwrap_or_else(|_| "https://metrics.solana.com:8086".to_string());
```
Comment (mvines) [inline src/metrics.rs]: dedup with https://github.com/solana-labs/solana/blob/4196cf43e84139170991c9a48c9dba2f67864801/src/metrics.rs#L40-L44 please

File: src/bin/upload-perf.rs [hunk 5]
```diff
@@ -18,33 +40,77 @@ fn main() {
...
         .output()
         .expect("failed to execute git rev-parse");
     let git_commit_hash = String::from_utf8_lossy(&git_output.stdout);
     let trimmed_hash = git_commit_hash.trim().to_string();
-    println!("uploading hash: {}", trimmed_hash);
+
+    let mut last_commit = None;
+    let mut results = HashMap::new();
+
+    let db = env::var("INFLUX_DATABASE").unwrap_or_else(|_| "scratch".to_string());
```
Comment (mvines) [inline src/bin/upload-perf.rs]: darn, too bad we need to keep this.  owell

File: src/metrics.rs (hunk 4 above)
Comment (pgarg66) [reply src/metrics.rs]: Would adding these as fields to structure, and setting in new() be the correct change?

Comment (pgarg66) [reply src/metrics.rs]: Or, maybe not, as query() doesn't create an instance of InfluxDbMetricsWriter..

Comment (mvines) [reply src/metrics.rs]: Could just have a static helper function that fetches the values, used by both places

**Pull request merged**